endorsement/storeapi.proto

```diff
@@ -43,6 +43,76 @@ message OpenArgs {
 
 message CloseArgs {
 }
+message SwComponentID {
+    common.TokenFormat type = 1;
+    google.protobuf.Struct id = 2;
+}
+// SwComponentInfo defines the structure of a single Software Component 
+// identified by a SoftwareComponent Identifer.
+message SwComponentInfo {
+    SwComponentID comp_id = 1;
+    google.protobuf.Struct comp_meta = 2; 
+}
+
+message AddSwComponentArgs {
+    repeated SwComponentInfo info = 1;
+}
+
+message GetSwComponentArgs {
+    string software_comp_id = 1;
+}
+
+message GetSwComponentRsp {
+    SwComponentInfo sw_component = 1;
+    int32 error_value = 2;
+    string error_detail = 3;
+}
+
+enum TaType {
+    UNSPEICIFIED = 0;
+    KEY = 1;
+    CERTIFICATE = 2;
+    INVALID = 3;
```

Comment:
if we want this to be extensible we should move INVALID to the end of the type range